test/e2e/kamsuite/kamsuite.go

```diff
@@ -284,10 +282,16 @@ func argocdAPIServer() (string, error) {
 	deployments := []string{"openshift-gitops-server", "openshift-gitops-repo-server",
 		"openshift-gitops-redis", "openshift-gitops-applicationset-controller", "kam", "cluster"}
 
-	for deployment := range deployments {
-		if waitForDeploymentsUpAndRunning("openshift-gitops", deployments[deployment]); err != nil {
-			return "", err
+	err = wait.Poll(time.Second*1, time.Minute*10, func() (bool, error) {
```

Comment:
I think we should wait for individual components. Because even if a condition fails in the poll function, we don't have to go through the loop again. WDYT?